.github/workflows/version-manifest.yml

```diff
@@ -0,0 +1,40 @@
+name: "Version manifest"
+
+# Gets the following content from the `VERSION` manifest:
+# 1. current version: the target version for the Terraform plan and apply.
+# 2. previous version: the previously deployed Terraform version.
+
+on:
+  workflow_call:
+    outputs:
+      previous:
+        description: "The previous contents of the VERSION manifest"
+        value: infrastructure/${{ jobs.version-manifest.outputs.previous }}
+      current:
+        description: "The current contents of the VERSION manifest"
+        value: infrastructure/${{ jobs.version-manifest.outputs.current }}
+
+jobs:
+  version-manifest:
+    runs-on: ubuntu-latest
+    outputs:
+      previous: ${{ steps.versions.outputs.previous }}
+      current: ${{ steps.versions.outputs.current }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Versions
+        working-directory: "infrastructure"
+        id: versions
+        run: |
+          PREV_SHA="$(git show -2 --pretty=format:"%h" --no-patch VERSION | tail -n 1)"
+          PREV_VERSION="$(git show ${PREV_SHA}:infrastructure/VERSION)"
+          CURR_VERSION="$(cat VERSION)"
+          echo "PREV_SHA=${PREV_SHA}"
+          echo "PREV_VERSION=infrastructure/${PREV_VERSION}"
+          echo "CURR_VERSION=infrastructure/${CURR_VERSION}"
+          echo "::set-output name=previous::v${PREV_VERSION}"
+          echo "::set-output name=current::v${CURR_VERSION}"
```

Comment:
Since you're using the pattern `infrastructure/v1.1.1` for your infrastructure changes, I think it would make sense here to return the full tag name in the output like so:
```sh
echo "::set-output name=previous::infrastructure/v${PREV_VERSION}"
echo "::set-output name=current::infrastructure/v${CURR_VERSION}"
```

That way you'll be able to use the output to directly compare and checkout tags.